source/JarResolverLib/src/Google.JarResolver/PlayServicesSupport.cs

```diff
@@ -1038,7 +1036,10 @@ public void ClearDependencies()
                 KeyValuePair<Dependency, string> oldDepFilenamePair;
                 if (currentDepsByVersionlessKey.TryGetValue(dep.VersionlessKey,
                                                             out oldDepFilenamePair)) {
-                    if (oldDepFilenamePair.Key.BestVersion != dep.BestVersion &&
+                    string oldVersion = ExtractVersionFromFileName(
+                        oldDepFilenamePair.Key.BestVersion);
+                    string newVersion = ExtractVersionFromFileName(dep.BestVersion);
+                    if ((oldVersion == null || (newVersion != null && oldVersion != newVersion)) &&
                         (confirmer == null || confirmer(oldDepFilenamePair.Key, dep))) {
```

Comment:
The stack trace reported in bug #48 starts here and bubbles up through to Resolver1_1.AarPathToPackageName where - from your report - the filename was null and therefore caused a null reference exception.  Which suggests https://github.com/googlesamples/unity-jar-resolver/blob/master/source/PlayServicesResolver/src/ResolverVer1_1.cs#L723 is returning a null path (as artifactPath) instead.